File: app/src/main/java/com/cazimir/relaxoo/model/AboutItem.java
package com.cazimir.relaxoo.model;

public class AboutItem {
    private int icon;
    private String name;

    private AboutItem(int icon, String name) {
        this.icon = icon;
        this.name = name;
    }

    public int getIcon() {
        return icon;
    }

    public String getName() {
        return name;
    }

    public static final class AboutItemBuilder {
        private int icon;
        private String name;

        public static AboutItemBuilder anAboutItem() {
            return new AboutItemBuilder();
        }

        public AboutItemBuilder withIcon(int icon) {
            this.icon = icon;
            return this;
        }

        public AboutItemBuilder withName(String name) {
            this.name = name;
            return this;
        }

        public AboutItem build() {
            return new AboutItem(this.icon, this.name);
        }
    }
}

File: app/src/main/java/com/cazimir/relaxoo/model/AboutItem.kt
package com.cazimir.relaxoo.model

class AboutItem(val name: AboutItemType, val icon: Int)

sealed class AboutItemType(val textToDisplay: String) {
    class SendFeedback(val text: String) : AboutItemType(text)
    class RemoveAds(val text: String) : AboutItemType(text)
    class Share(val text: String) : AboutItemType(text)
    class PrivacyPolicy(val text: String) : AboutItemType(text)
    class RateApp(val text: String) : AboutItemType(text)
    class MoreApps(val text: String) : AboutItemType(text)
}

